ECPLiBase.py

```diff
@@ -58,29 +58,29 @@ class ECPLiBase(ABC):
        Attributes:
             limit_target: Description of the parameter for which a limit is to
                           be derived.
-            dataset: Actuall gamma-ray data in form of a
-                  gammapy.modeling.Dataset.
-                  This can in practice e.g. be a MapDataset (3d-analysis)
-                  or a SpectrumDataset (1d-analysis).
+            datasets: Actuall gamma-ray data in form of a
+                  gammapy.modeling.Datasets.
+                  This can in practice e.g. be a MapDatasets (3d-analysis)
+                  or a SpectrumDatasets (1d-analysis).
             CL: Confidence level on which to work.
             ul: Upper limit on the parameter speficied by limit_target at the
                 confidence level given in the class constructor.
     """
     def __init__(self,
                  limit_target: LimitTarget,
-                 dataset: Dataset,
+                 datasets: Datasets,
                  CL: float):
 
         self.limit_target = limit_target
-        self.dataset = dataset.copy()
+        self.datasets = datasets.copy()
```

Comment:
the copy will be inefficient for large datasets and could likely be avoided as modifications of the original datasets that affect only parameters values/frozen status can be reverted with a context manager as https://docs.gammapy.org/0.18.2/api/gammapy.modeling.Parameters.html#gammapy.modeling.Parameters.restore_values

Reply:
I know. Didn't have the time for optimizations.